chrome/content/test/README.md

```diff
@@ -1,24 +1,145 @@
+
+A `coverage` function is defined for every module(for which unit tests exist). This `coverage` function returns the coverage info for the corresponding module. 
+
+The coverage info exists in a variable in the generated instrumented module.The variable name is generated by following the same [steps](https://github.com/gotwarlost/istanbul/blob/master/lib/instrumenter.js#L43) as done by `istanbul`.
+
+To generate the coverage variabe name, we need the full file path and a hashing function. The hashing fuction is built using [nsICryptoHash](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsICryptoHash#Computing_the_Hash_of_a_String) and the [base64](https://github.com/Gozala/crypto/blob/master/md5.js#L90) implementation extracted from the node [Crypto](https://github.com/Gozala/crypto) module(since `istanbul` [uses](https://github.com/gotwarlost/istanbul/blob/master/lib/instrumenter.js#L13) it).
+
+The coverage variable name is a md5 sum + base64 encryption of the full file path. The `pwd` directory path is passed in `test.json` by the `run_test.sh` bash script. This is an automated process and developers should not edit `test.json` manually. The full file path is put together here.
+
+#### Collecting the coverage info from every module ####
 
-The CommonJS extension modules are loaded in [chrome/content/test/globals.js](https://github.com/privly/privly-jetpack/tree/master/chrome/content/test/globals.js) and are exposed to the tests via a global `g` namespace. The extension modules are loaded via a custom XPCOM object defined in [lib/xpcom.js](https://github.com/privly/privly-jetpack/blob/master/lib/xpcom.js).
+Once all the tests have run, we loop over all the modules used in the tests and call the `coverage` function. Once we have the coverage info from all the files, we send it to a nodejs server running locally that accepts this coverage info and generates a report. This report can be seen on the terminal when the tests are run on a local machine. If these tests are run on Travis-CI, then the generated report is sent to Coveralls.io as well.
 
-To run the tests for your platform, add `--prefs test.json` to the jpm run command.
+#### Running a NodeJS Server that accepts the coverage info and sends it to Coveralls.io ####
 
-Adding the `--prefs` arg tells the extension to run the unit-tests, and opens up the `test_loader.html`. The code to do so can be found in [lib/run_test.js](https://github.com/privly/privly-jetpack/blob/master/lib/run_test.js).
+The server is run using Node.js. We've used Node.JS because this makes it easier to generate the reports from the coverage json and send those reports to Coverall.io. `istanbul` node [API](https://github.com/gotwarlost/istanbul#generate-reports-given-a-bunch-of-coverage-json-objects) generates the reports from the coverage json and [node-coveralls](https://github.com/nickmerwin/node-coveralls) sends those reports to Coveralls.io.
```

Comment:
Specifying that the name of the server is `server.js` would answer some questions a newcomer would have.